### PR TITLE
Initial schema and testing

### PR DESCRIPTION
**Pull request by jkeifer**

### CODE REVIEW

File: db/migrations/20230412000000_base.sql [hunk 1]
```diff
@@ -0,0 +1,452 @@
...
+),
+('INFO', 'Event is informational and does not change thread state');
+
+
+CREATE TABLE IF NOT EXISTS swoop.action (
+  action_uuid uuid NOT NULL DEFAULT gen_random_uuid(),
+  action_type text NOT NULL CHECK (action_type IN ('callback', 'workflow')),
+  action_name text NOT NULL,
+  parent_uuid bigint, -- reference omitted, we don't need referential integrity
+  workflow_name text,  -- not explicitly required, but helpful for queries
```
Comment (jkeifer) [inline db/migrations/20230412000000_base.sql:42]: remove comment

Comment (jkeifer) [reply db/migrations/20230412000000_base.sql:42]: rename to `name`, make it optional for callbacks?

We need some examples of how this it expected to work for workflows vs callbacks.

Comment (jkeifer) [reply db/migrations/20230412000000_base.sql:42]: @matthewhanson After some reflection, perhaps these fields should be named like:

* `action_name` -> `handler_name`
* `workflow_name` -> `action_name`

I'll use the latter names for consistency in the following explanation.

The idea is that an action is a discreet, independent thing that needs to be acted upon in some way. `action_type` is to indicate whether the action is a workflow or a callback. At the moment there isn't much difference between how these two types would be handled, but is required to separate workflow actions (jobs to the API) from their associated callback actions.

`handler_name` is effectively designates a specific queue with rate limiting, error handling, endpoint, and other such configuration. `swoop-conductor` will need to be configured explicitly with those details for each handler that it is servicing. `handler_name` will also be used as the `NOTIFY` channel name, so `swoop-conductor` instances will only get notification for actions associated with the handlers each has configured.

`action_name` is again not really part of how the action is handled, but is necessary to allow the API to group by workflow names. Callbacks don't necessarily need a name in the way workflow actions will, but we can make the field generic and then optional for callbacks, so it will be available if we do have a use for it in the future.

Let's see an example to pull this all together.

Say we have a workflow backed by Argo Workflows named `process-data`. Let's say it creates one output item and that item should have one publish callback action spawned on success. In this case we get two rows that are like:

```
action_type: workflow
handler_name: argo-workflow
action_name: process-data
```

and

```
action_type: callback
handler_name: publish-s3
action_name: <null>
```

In short, `swoop-conductor` really only needs to know the handler name to handle the action. The other two fields are essentially there as grouping attributes for the API and operational queries.

**Review by paulpilone — APPROVED** [review]

**Review by matthewhanson — APPROVED** [review]

File: .env
Comment (paulpilone) [inline .env:1]: @jkeifer do we want this file at the root directory? When I'm configuring the settings by default pydantic only looks for an .env in the working directory (I'll test passing a full path - so maybe we can tell it to look in the root project directory).

File: .env [hunk 2]
```diff
@@ -0,0 +1,7 @@
+DATABASE_HOST="localhost"
+DATABASE_PORT="5432"
+DATABASE_USER="postgres"
+DATABASE_PASS="password"
+DATABASE_NAME="swoop"
+DBMATE_MIGRATIONS_TABLE="swoop.schema_migrations"
+DATABASE_URL="postgres://${DATABASE_USER}:${DATABASE_PASS}@${DATABASE_HOST}:${DATABASE_PORT}/${DATABASE_NAME}?sslmode=disable"
```
[the remark this answers is not in the document]
Comment (jkeifer) [reply .env:7]: These vars are not just used by `swoop-api` but also docker and the db tooling.

File: .env
Comment (jkeifer) [reply .env:1]: I think generally best practice is to have your `.env` files in the root of your project.
